src/main/java/seedu/address/model/person/Role.java

```diff
@@ -5,13 +5,25 @@
  * Provides utility methods for converting a string into a Role and for validating if a string is a valid Role.
  */
 public enum Role {
-    VOLUNTEER,
-    DONOR,
-    PARTNER,
-    PERSON;
+    VOLUNTEER(Volunteer.class),
```

Comment:
Nice work done here to tie roles to their related classes! (Just trying to meet the PR review requirement)